submissions/Dovahkiin1991/classic-frogger-game/app.js

```diff
@@ -0,0 +1,124 @@
+//const
+const CELL_WIDTH = 101,
+    CELL_HEIGHT = 83,
+    ALL_ENEMIES = [50, 60, 30];
+const Enemy = function(y, index, player) {
+    this.sprite = 'images/enemy-bug.png';
+    this.x = -CELL_WIDTH;
+    this.y = CELL_HEIGHT - 23 + index * CELL_HEIGHT;
+    this.speed = speedGenerator();
+    this.player = player;
+};
+
+Enemy.prototype.update = function(dt) {
+    this.x += dt * this.speed;
+
+    if (this.x > 505) {
+        this.x = -CELL_WIDTH;
+        this.speed = speedGenerator();
+    }
+
+    this.collisionsCheck();
+};
+
+Enemy.prototype.collisionsCheck = function() {
```

Comment:
`player` in this method is a global variable.
See item 4.iii of the [requirements](https://kottans.org/documentation/docs/doc/code-review/#oo-js-frogger).